packages/lisk-p2p/src/peer_pool.ts

```diff
@@ -14,107 +14,47 @@
  */
 
 /**
- * The purpose of the PeerPool
+ * The purpose of the PeerPool is to provide a simple interface for selecting,
+ * interacting with and handling aggregated events from a collection of peers.
  */
 /* tslint:disable:variable-name */
 import { EventEmitter } from 'events';
-import http, { Server } from 'http';
-import querystring from 'querystring';
-import socketClusterServer from 'socketcluster-server';
-import { PeerTransportError } from './errors';
 import { Peer } from './peer';
 
-export const EVENT_INBOUND_PEER_FAIL = 'inboundPeerFail';
-export const EVENT_NEW_INBOUND_PEER = 'newInboundPeer';
-export const EVENT_NEW_PEER = 'newPeer';
-
-export interface PeerPoolConfig {
-	readonly blacklistedPeers?: ReadonlyArray<string>;
-	readonly connectTimeout: number;
-	readonly ipAddress?: string;
-	readonly seedPeers: ReadonlyArray<string>;
-	readonly wsEngine?: string;
-	readonly wsPort: number;
-}
-
 export class PeerPool extends EventEmitter {
```

Comment:
does at any point later `EventEmitter` be used here? or it will be used by`P2P` as now it has reference to `PeerPool` instance, maybe we can remove it

Reply:
Based on our last discussion, the current plan was for the peer to emit events directly on the PeerPool instead of making the PeerPool add event handlers to every peer and then manually cleaning from every peer.

Reply:
Based on our new discussion, I'm going for the approach of not passing the PeerPool to the Peer object and will make the PeelPool add and manage the the event handlers on the peer object.